### PR TITLE
Add collect limits.

### PR DESCRIPTION
**Pull request by briancao**
Initial COLLECT limit.

### CODE REVIEW

File: pages/api/collect.ts [hunk 1]
```diff
@@ -88,7 +104,21 @@ export default async (req: NextApiRequestCollect, res: NextApiResponse) => {
 
   await useSession(req, res);
 
-  const session = req.session;
+  const { session, website } = req.session;
+
+  // Check collection limit
+  if (process.env.ENABLE_COLLECT_LIMIT) {
+    const userId = website.userId ? website.userId : website.team.userId;
```
Comment (mikecao) [inline pages/api/collect.ts:111]: The team user id does not mean they are the owner. The role defines the owner.

File: lib/session.ts [hunk 2]
```diff
@@ -85,12 +106,12 @@ export async function findSession(req) {
...
-    } catch (e) {
+    } catch (e: any) {
       if (!e.message.toLowerCase().includes('unique constraint')) {
         throw e;
       }
     }
   }
 
-  return session;
+  return { session, website };
```
Comment (mikecao) [inline lib/session.ts:116]: We should not overload this method. For our purpose it's cached anyways. Better to keep the logic clear.

**Review by mikecao — CHANGES_REQUESTED** [review]
A majority of this logic should not be in the app. Only we care about limits, the end user never will. The app should only care if a number is hit, then abort.
The real logic will either live on cloud side or in cron scripts that populate redis caches, etc.

File: lib/session.ts (hunk 2 above)
Comment (briancao) [reply lib/session.ts:116]: Then we have to make a 2nd call to the cached website, outside of this method then.  It's not a big most of the time, but during high traffic scenarios, we should be squeezing as much performance/efficiency as we can out of the collect.

File: pages/api/collect.ts (hunk 1 above)
Comment (briancao) [reply pages/api/collect.ts:111]: > A majority of this logic should not be in the app. Only we care about limits, the end user never will. The app should only care if a number is hit, then abort. The real logic will either live on cloud side or in cron scripts that populate redis caches, etc.

Right, I imagine cloud, will populate the 'user max' number through redis, but those membership total mechanisms isn't built yet.  I can make assumptions for it now though.

Comment (briancao) [reply pages/api/collect.ts:111]: We should get rid of user_id on the TEAM table if it doesn't represent owner anymore.

**Pull request closed without merging**